src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/OldGenRca.java

```diff
@@ -252,40 +254,31 @@ public double readMin() {
     }
 
     /**
-     * Sliding window to check the max/min old gen usage within a given time frame
+     * Sliding window to check the max/min olg gen usage within a given time frame
      *
      * @param isMinSlidingWindow true if the sliding window is for min usage, false for max usage
      *     Provides a more general framework than MinOldGenSlidingWindow as this sliding window can
      *     be implemented as minSlidingWindow or maxSlidingWindow depending on the need.
      */
     public static class MinMaxSlidingWindow extends SlidingWindow<SlidingWindowData> {
-        boolean isMinSlidingWindow;
+        BiConsumer<Deque<SlidingWindowData>, SlidingWindowData> nextElementFunc;
 
         public MinMaxSlidingWindow(
                 int SLIDING_WINDOW_SIZE_IN_TIMESTAMP,
                 TimeUnit timeUnit,
-                boolean isMinSlidingWindow) {
+                BiConsumer<Deque<SlidingWindowData>, SlidingWindowData> nextElementFunc) {
             super(SLIDING_WINDOW_SIZE_IN_TIMESTAMP, timeUnit);
-            this.isMinSlidingWindow = isMinSlidingWindow;
+
+            // get the Biconsumer lambda function passed in
+            this.nextElementFunc = nextElementFunc;
         }
 
         @Override
         public void next(SlidingWindowData e) {
-            if (isMinSlidingWindow) {
-                // monotonically decreasing sliding window
-                while (!windowDeque.isEmpty()
-                        && windowDeque.peekFirst().getValue() >= e.getValue()) {
-                    windowDeque.pollFirst();
-                }
-            } else {
-                // monotonically increasing sliding window
-                while (!windowDeque.isEmpty()
-                        && windowDeque.peekFirst().getValue() < e.getValue()) {
-                    windowDeque.pollFirst();
-                }
-            }
+            // use the passed in lambda function to accept next element
+            nextElementFunc.accept(windowDeque, e);
 
```

Comment:
Why do we need this change in [OldGenRca.java](https://github.com/opensearch-project/performance-analyzer-rca/pull/461/files#diff-51eda0cc53b9494933631d232c7ca0c1570151d9f65b0fba48bffe1f2456752b) ?

Reply:
We change this part according to Open-Close Principle.
It opens `extension` for downstream class to customize the lambda function to accept next elements, and I made corresponding changes in SearchBackPressureRCA to add the lambda function.